src/App.js

```diff
@@ -1,6 +1,74 @@
 import React from 'react';
-import './App.css';
+import { Pagination } from './components/Pagination';
 
-const App = () => <h1>Pagination</h1>;
+import './App.scss';
+
+class App extends React.PureComponent {
+  state = {
+    selectedPage: 1,
+    perPage: 5,
+  }
+
+  changePage = pageNumber => (
+    // eslint-disable-next-line consistent-return
+    this.setState((state) => {
+      if (pageNumber !== state.selectedPage) {
+        return {
+          selectedPage: pageNumber,
+        };
+      }
+    })
+  )
```

Comment:
You are not using previous state directly in calculations, so it is possible to use simple setState here. And try to avoid disabling eslint, this is final measure to use.

```suggestion
  changePage = pageNumber => {
    const { selectedPage } = this.state;
    
    if (pageNumber !== selectedPage) {
      this.setState({
        selectedPage: pageNumber,
      });
    }
  }
```